### PR TITLE
Updates for selection sort

### LINKED ISSUE #19: Challenge: selection_sort

### PR DESCRIPTION
**Pull request by mewael-alema**
- [ ] PR's title includes the challenge's name and language
- [ ] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [ ] the branch is up to date with `main`/`master`
- [ ] the code works when pulled and run locally
- [ ] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [x] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [x] The markdown looks correct when you preview the file
- [x] All links and images work
- [x] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [x] There is a module header
- [ ] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [ ] Behavior description
- [ ] Parameter description
- [ ] Return value description
- [ ] Include any assumptions
- [ ] Include 3 or more (passing!) doctests
- [ ] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [ ] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [x] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [x] There are type annotations
- [x] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Each unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [x] Only one assertion
- [ ] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases
- [x] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/sorting/python/selection_sort/src/selection_sort.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def selection_sort(num_array):
+	array_len = len(num_array)
+	for i in range(array_len - 1):
+		for j in range(i + 1, array_len):
+			if (num_array[j] < num_array[i]):
+				temp = num_array[i]
+				num_array[i] = num_array[j]
+				num_array[j] = temp
+    return num_array
```
Comment (VolodymyrYakovenko) [inline src/sorting/python/selection_sort/src/selection_sort.py:9]: image

**Review by VolodymyrYakovenko — CHANGES_REQUESTED** [review]
All necessary edits are indicated on the sheet in description - unchecked checkboxes. 

Your branch is NOT up-to-date with main branch. 

Traceback also appears when I'm trying to compile your solution.
